reactioncommerce/reaction
# feat: add ordersByAccountId query

Resolves #4579
Impact: **minor**  
Type: **feature**

## Issue
While we had `ordersByAccountId` outlined in our GraphQL schema, we had not yet built out the query to use this.

## Solution
Add the needed query to be able to use `ordersByAccountId`

## Testing
1. Create orders from multiple accounts
1. Add the following into the `Shops` collection, if you'd like to see translations of order statuses
```
"orderStatusLabels": {
    "new" : [ 
        {
            "language" : "en",
            "label" : "Order received"
        }, 
        {
            "language" : "es",
            "label" : "Recibido"
        }
    ],
    "coreOrderWorkflow/processing" : [ 
        {
            "language" : "en",
            "label" : "Order received"
        }, 
        {
            "language" : "es",
            "label" : "Recibido"
        }
    ],
    "coreOrderWorkflow/completed" : [ 
        {
            "language" : "en",
            "label" : "Shipped"
        }, 
        {
            "language" : "es",
            "label" : "Enviado"
        }
    ],
    "coreOrderWorkflow/canceled" : [ 
        {
            "language" : "en",
            "label" : "Canceled"
        }, 
        {
            "language" : "es",
            "label" : "Cancelado"
        }
    ]
}
```
1. Use the following `query`, and make sure you only get back orders from the correct supplied accountId

```
{
  ordersByAccountId(accountId: "PUT_ACCOUNT_ID_HERE") {
    edges {
      node {
        _id
        status(language: "es") {
	  label
          status
        }
        ...anyOtherFieldsYouWantToTest
      }
    }
  }
}
```

```
{
  ordersByAccountId(accountId: "PUT_ACCOUNT_ID_HERE", shopIds: ["PUT_SHOP_ID(s)_HERE]) {
    edges {
      node {
        _id
        status(language: "es") {
	  label
          status
        }
        ...anyOtherFieldsYouWantToTest
      }
    }
  }
}
```

1. Remove translations and make sure query still works and returns raw `status` data if translations are not available at all (not in collection), or for the requested language (request `fr`)
1. If you'd like to see this in the UI, you can test against this starterkit branch: https://github.com/reactioncommerce/reaction-next-starterkit/pull/393 - however that one isn't quite ready so any UI errors please ignore, only use it to visualize the API as well as it can do for now
